rzup/src/cli/commands.rs

```diff
@@ -0,0 +1,300 @@
+        }
+        println!("{}", "Installed components:".bold());
+        println!("{}", "--------------------".bold());
+
+        let mut component_ids: Vec<_> = components
+            .iter()
+            .filter(|c| c.parent_component().is_none())
+            .map(|c| c.id())
+            .collect();
+        component_ids.sort_unstable();
+
+        for component_id in component_ids {
+            let versions = rzup.list_versions(component_id)?;
+            if !versions.is_empty() {
+                println!("\n{}", component_id);
+
+                let active_version = rzup.get_active_version(component_id)?;
+                let current_version = rzup.settings().get_active_version(component_id);
+
+                let mut sorted_versions = versions.clone();
+                sorted_versions.sort_by(|a, b| b.cmp(a)); // sort newest to oldest
+
+                for version in sorted_versions {
+                    let is_active = active_version
+                        .as_ref()
+                        .map_or(false, |(v, _)| v == &version);
+                    let marker = if is_active { "* " } else { "  " };
+                    println!("{}{}", marker.bold(), version);
+                }
+
+                // Only show warning if version in settings doesn't exist in versions list
+                if let Some(settings_version) = current_version {
+                    if !versions.contains(&settings_version) {
+                        println!(
+                            "! Version {} specified in settings.toml is not installed",
+                            settings_version
+                        );
+                        println!(
+                            "  Please use 'rzup use {} <VERSION>' to switch active component",
+                            component_id,
+                        );
+                    }
+                }
+            }
+        }
+        println!(
+            "\n{}: {}",
+            "rzup home".bold(),
+            rzup.environment.root_dir().display()
+        );
+        Ok(())
+    }
+}
+
+#[derive(Parser)]
+pub(crate) struct UseCommand {
+    /// Name of component to activate
+    #[arg(value_parser=component_parser())]
+    name: String,
+    /// Version of component to activate
+    version: String,
+}
+
+impl UseCommand {
+    pub(crate) fn execute(self, rzup: &mut Rzup) -> Result<()> {
+        let version = Version::parse(&self.version)
+            .map_err(|_| RzupError::InvalidVersion(self.version.clone()))?;
+
+        if rzup.version_exists(&self.name, &version)? {
+            rzup.set_active_version(&self.name, version.clone())?;
+            println!(
+                "Successfully set {} version {} as active",
+                self.name, version
+            );
+        } else {
+            println!(
+                "! Version {} of {} is not installed.\n  Please use 'rzup install {} {}' to install",
+                version, self.name, self.name, version
```

Comment:
Here and other places you can also write this sort of thing like this:
```
println!(
    "! Version {version} of {name} is not installed.\n  Please use 'rzup install {name} {version}' to install",
    name=self.name
);
```

Reply:
If the format string is a little long it can be a littler easier to understand whats being printed if you do it this way